### PR TITLE
[ads] Allow non-Rewards users to clear `Brave Ads data` from `Delete browsing data`

### PR DESCRIPTION
**Pull request by aseren**
Enabled clearing of `Brave Ads data` from the settings for non-Rewards users. Clearing of Brave Ads data deletes/clears the following data in users Profile:
- **Default/ads_service/database.sqlite**
- **Default/ads_service/client.json**
- **Default/ads_service/confirmations.json**
- **Default/Preferences/brave/brave_ads**

**On Desktop**:
Option is available in `Delete browsing data` dialog for non-Rewards users. If user has joined Brave Rewards then it is replaced by `Reset rewards data...` link, which leads to brave://rewards/#reset.
image

**On iOS**:
Option is available in `Brave Shields & Privacy` dialog non-Rewards users. If user enabled Brave Rewards, the this option is hidden.
image

Resolves https://github.com/brave/brave-browser/issues/39051

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Test case 1 (Desktop)
* Fresh install
* Trigger search result ad for a staging server with the `giraffe` request
* Click search result ad
* Convert search result ad by visiting https://www.brave.com/vertical-tabs/
* Check that following tables in `ads_service/database.sqlite` are not empty:
    - ad_events
    - creative_set_conversions
* Open `Settings` -> `History` -> `Delete browsing data`
* Click `Clear Brave Ads data...` link
* Make sure that Brave Ads service is restarted
* Check that following tables in `ads_service/database.sqlite` are empty:
    - ad_events
    - creative_set_conversions
    - confirmation_queue
    - transactions
* Check that `ads_service/client.json` contains json with empty fields
* Check that `ads_service/confirmations.json` contains json with empty fields

### Test case 2 (Desktop)
* Fresh install
* Join Brave News
* Trigger search result ad for a staging server with the `giraffe` request
* Click search result ad
* Convert search result ad by visiting https://www.brave.com/vertical-tabs/
* View/click/land Brave News ad
* Convert Brave News ad
* Check that following tables in `ads_service/database.sqlite` are not empty:
    - ad_events
    - creative_set_conversions
* Open `Settings` -> `History` -> `Delete browsing data`
* Click `Clear Brave Ads data...` link
* Make sure that Brave Ads service is restarted
* Check that following tables in `ads_service/database.sqlite` are empty:
    - ad_events
    - confirmation_queue
    - transactions
* Check that `creative_set_conversions` table doesn't contain row with creative_set_id =  `d81220f2-b7c6-4f3e-b3fa-5e0f579407a5`
* Check that `ads_service/client.json` contains json with empty fields
* Check that `ads_service/confirmations.json` contains json with empty fields

### Test case 3 (Desktop)
* Fresh install
* Join Brave Rewards
* Open `Settings` -> `History` -> `Delete browsing data`
* Make sure `Reset Brave Rewards data...` link is shown

### Test case 4 (iOS)
* Fresh install
* Trigger search result ad for a staging server with the `giraffe` request
* Click search result ad
* Convert search result ad by visiting https://www.brave.com/vertical-tabs/
* Check that following tables in `ads/Ads.sqlite` are not empty:
    - ad_events
    - creative_set_conversions
* Toggle `Settings` -> `Shields & Privacy` -> `CLEAR PRIVATE DATA` -> `Brave Ads Data`
* Tap `Clear Data Now` button
* Make sure that Brave Ads service is restarted
* Check that following tables in `ads/Ads.sqlite` are empty:
    - ad_events
    - creative_set_conversions
    - confirmation_queue
    - transactions
* Check that `ads/client.json` contains json with empty fields
* Check that `ads/confirmations.json` contains json with empty fields

### Test case 5 (iOS)
* Fresh install
* Join Brave News
* Trigger search result ad for a staging server with the `giraffe` request
* Click search result ad
* Convert search result ad by visiting https://www.brave.com/vertical-tabs/
* View/click/land Brave News ad
* Convert Brave News ad
* Check that following tables in `ads/Ads.sqlite` are not empty:
    - ad_events
    - creative_set_conversions
* Toggle `Settings` -> `Shields & Privacy` -> `CLEAR PRIVATE DATA` -> `Brave Ads Data`
* Tap `Clear Data Now` button
* Make sure that Brave Ads service is restarted
* Check that following tables in `ads/Ads.sqlite` are empty:
    - ad_events
    - confirmation_queue
    - transactions
* Check that `creative_set_conversions` table doesn't contain row with creative_set_id =  `d81220f2-b7c6-4f3e-b3fa-5e0f579407a5`
* Check that `ads/client.json` contains json with empty fields
* Check that `ads/confirmations.json` contains json with empty fields

### Test case 6 (iOS)
* Fresh install
* Join Brave Rewards
* Go to `Settings` -> `Shields & Privacy` -> `CLEAR PRIVATE DATA`
* Make sure `Brave Ads Data` toggle is not visible

### CODE REVIEW

File: ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift [hunk 1]
```diff
@@ -192,6 +192,14 @@ public class BraveRewards: PreferencesObserver {
     }
   }
 
+  // MARK: - Brave Ads History
+
+  /// Clear Brave Ads History.
+  @discardableResult
+  func clearAdsHistory() async -> Bool {
+    return await ads.clearHistory()
+  }
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift:201]: This will run off the UI task runner/thread, if it needs to be run on main please add `@MainActor` alongside `@discardableResult`

File: ios/browser/api/ads/brave_ads.mm [hunk 2]
```diff
@@ -1785,6 +1785,17 @@ - (void)toggleDislikeAd:(NSString*)creativeInstanceId
...
 
+- (void)clearHistory:(void (^)(BOOL success))completion {
+  if (![self isServiceRunning]) {
+    return completion(/*success=*/false);
+  }
+
+  ads->ClearHistory(base::Time(), base::Time::Max(),
+                    base::BindOnce(^(const bool success) {
+                      completion(success);
+                    }));
```
Comment (kylehickinson) [inline ios/browser/api/ads/brave_ads.mm:1796]: You may be able to shorten this to `base::BindOnce(completion)` depending on if clang considers `bool` and `BOOL` different types

File: ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift (hunk 1 above)
Comment (tmancey) [reply ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift:201]: This is specifically to clear the history of Brave Ads, as these ads can still run even if Rewards are disabled for search result ads.

Comment (kylehickinson) [reply ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift:201]: Assume this was meant to be commented on the above review comment but I just meant will a user even understand what this means given that we dont really have the words "Brave Ads" written anywhere in the app

Comment (tmancey) [reply ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift:201]: Ah yes, let me move this to that thread. Thanks

File: ios/brave-ios/Sources/BraveStrings/BraveStrings.swift [hunk 3]
```diff
@@ -9191,6 +9191,16 @@ extension Strings {
       value: "Brave Rewards",
       comment: ""
     )
+
+    public static let braveAdsHistoryToggleOption =
+      NSLocalizedString(
+        "ads.braveAdsHistoryToggleOption",
+        tableName: "BraveShared",
+        bundle: .module,
+        value: "Brave Ads History",
```
[the remark this answers is not in the document]
Comment (tmancey) [reply ios/brave-ios/Sources/BraveStrings/BraveStrings.swift:9200]: This is specifically to clear the history of Brave Ads, because the ads component can still run for search result ads even if Rewards are disabled.

[the remark this answers is not in the document]
Comment (tmancey) [reply ios/brave-ios/Sources/BraveStrings/BraveStrings.swift:9200]: @fmarier your thoughts, thanks

[the remark this answers is not in the document]
Comment (tmancey) [reply ios/brave-ios/Sources/BraveStrings/BraveStrings.swift:9200]: @fmarier on iOS they have "Brave News", so for consistency should your thoughts on "Brave Ads", instead of "Brave Ads History"?

File: ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift (hunk 1 above)
Comment (tmancey) [reply ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift:201]: @kylehickinson maybe we could add a description for this switch? Thanks

File: ios/browser/api/ads/brave_ads.mm (hunk 2 above)
Comment (aseren) [reply ios/browser/api/ads/brave_ads.mm]: Fixed

File: ios/brave-ios/Sources/BraveStrings/BraveStrings.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (fmarier) [reply ios/brave-ios/Sources/BraveStrings/BraveStrings.swift:9200]: I'd suggest "Brave Ads data" (lower case *data*) on desktop for consistency with how the Chromium-inherited items look.

[the remark this answers is not in the document]
Comment (fmarier) [reply ios/brave-ios/Sources/BraveStrings/BraveStrings.swift:9200]: What does "Brave News" clear on iOS? The feed/image cache?

[the remark this answers is not in the document]
Comment (kylehickinson) [reply ios/brave-ios/Sources/BraveStrings/BraveStrings.swift:9200]: It deletes the downloaded feed & sources json files

File: components/brave_ads/browser/ads_service_impl.cc [hunk 4]
```diff
@@ -161,6 +161,19 @@ bool DeletePathOnFileTaskRunner(const base::FilePath& path) {
   return base::DeleteFile(path);
 }
 
+bool ClearHistoryOnExitOnFileTaskRunner(
+    const base::FilePath& ads_service_path,
+    const bool user_has_joined_brave_rewards) {
+  // Do not clear history for Rewards users from the database, as it is
```
Comment (fmarier) [inline components/brave_ads/browser/ads_service_impl.cc:167]: If we're not going to honor the user's request to delete Brave Ads data, then I think in that case the check box should be grayed out. Otherwise we're not doing what we "promised" to do by offering the checkbox.

File: components/brave_ads/core/internal/history/database_history_util_internal.cc [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+#include "brave/components/brave_ads/core/internal/user_engagement/ad_events/ad_event_cache_util.h"
+#include "brave/components/brave_ads/core/internal/user_engagement/ad_events/ad_events_database_table.h"
+#include "brave/components/brave_ads/core/mojom/brave_ads.mojom.h"
+
+namespace brave_ads {
+
+void MaybeClearDatabaseHistoryForDateRange(const base::Time from_time,
+                                           const base::Time to_time,
+                                           ResultCallback callback) {
+  // Do not clear history for Rewards users from the database, as it is
```
Comment (fmarier) [inline components/brave_ads/core/internal/history/database_history_util_internal.cc:25]: Same problem here: we can treat Rewards users differently given that they have opted in, but we need to make it clear whether or not data will be deleted.

File: ios/brave-ios/Sources/BraveStrings/BraveStrings.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (fmarier) [reply ios/brave-ios/Sources/BraveStrings/BraveStrings.swift:9200]: I'd suggest "Brave Ads Data" (uppercase *Data*) on iOS to match most of the other entries.

"Brave News" should probably also be "Brave News Data" (or maybe "Brave News Cache" which is more descriptive) though that's outside the scope of this issue.

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 6]
```diff
@@ -1263,6 +1263,10 @@ void RewardsServiceImpl::OnStopEngineForCompleteReset(
     const mojom::Result result) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
   profile_->GetPrefs()->ClearPrefsWithPrefixSilently(pref_prefix);
+  // Notify `brave_rewards::prefs::kEnabled` listeners that Brave Rewards was
+  // disabled.
+  profile_->GetPrefs()->SetBoolean(brave_rewards::prefs::kEnabled, false);
```
Comment (zenparsing) [inline components/brave_rewards/browser/rewards_service_impl.cc:1268]: Setting this value to a non-default value may cause the Rewards engine utility process to always start (see line 428). We definitely want to avoid that.

It might be enough to simply change the condition on line 428 to test the value of the "enabled" pref, rather than testing for a non-default value, but please test to make sure that the Rewards engine process starts and stops appropriately.

Comment (zenparsing) [reply components/brave_rewards/browser/rewards_service_impl.cc:1268]: Also, it's unfortunate that we can't simply clear prefs and be done. What part of the system are we trying to notify by setting the pref here?

File: browser/ui/webui/settings/brave_clear_browsing_data_handler.h [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  // ClearBrowsingDataHandler overrides:
+  void RegisterMessages() override;
+
+  void HandleGetBraveRewardsEnabled(const base::Value::List& args);
+
+  void HandleClearBraveAdsData(const base::Value::List& args);
+
+  void UpdateBraveRewardsEnabled();
+
+  raw_ptr<Profile> profile_;
```
Comment (aseren) [inline browser/ui/webui/settings/brave_clear_browsing_data_handler.h:35]: = nullptr;

File: browser/ui/webui/settings/brave_clear_browsing_data_handler.h [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+class BraveClearBrowsingDataHandler : public ClearBrowsingDataHandler {
+ public:
+  BraveClearBrowsingDataHandler(content::WebUI* webui, Profile* profile);
+  BraveClearBrowsingDataHandler(const BraveClearBrowsingDataHandler&) = delete;
+  BraveClearBrowsingDataHandler& operator=(
+      const BraveClearBrowsingDataHandler&) = delete;
+  ~BraveClearBrowsingDataHandler() override;
+
+ private:
+  // ClearBrowsingDataHandler overrides:
```
Comment (aseren) [inline browser/ui/webui/settings/brave_clear_browsing_data_handler.h:26]: // ClearBrowsingDataHandler:

File: components/brave_rewards/browser/rewards_service_impl.cc (hunk 6 above)
Comment (tmancey) [reply components/brave_rewards/browser/rewards_service_impl.cc:1268]: This change covers an edge case where the "Delete browsing data" modal would need to be on the screen at the same time Rewards was reset. This is an edge case and not worth the risk, or amount of effort for regression testing. I would recommend this change is removed because once the modal is closed and reopened the correct text would be shown.

File: ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController.swift [hunk 9]
```diff
@@ -376,10 +376,7 @@ public class BrowserViewController: UIViewController {
     // - Brave Rewards is enabled
     // - Brave News is enabled
     // - `ShouldAlwaysRunBraveAdsService` feature is enabled
```
Comment (tmancey) [inline ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController.swift:378]: This comment is not needed because it was moved to shouldStartAds

File: ios/browser/api/ads/brave_ads.mm [hunk 10]
```diff
@@ -1739,6 +1745,25 @@ - (void)toggleDislikeAd:(NSString*)creativeInstanceId
...
+  }
+
+  base::FilePath storage_path(base::SysNSStringToUTF8(self.storagePath));
+  base::ThreadPool::PostTaskAndReply(
+      FROM_HERE, {base::MayBlock()},
+      base::BindOnce(
+          [](base::FilePath storage_path) {
+            base::DeleteFile(storage_path.Append(kAdsDatabaseFilename));
+            base::DeleteFile(
+                storage_path.Append(brave_ads::kClientJsonFilename));
```
Comment (tmancey) [inline ios/browser/api/ads/brave_ads.mm:1761]: We should also delete confirmations.json on iOS because a user can toggle Rewards off (they cannot reset Rewards).

Comment (tmancey) [reply ios/browser/api/ads/brave_ads.mm:1761]: Why are we not clearing prefs?

File: browser/ui/webui/settings/brave_clear_browsing_data_handler.h [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+ private:
+  // ClearBrowsingDataHandler:
+  void RegisterMessages() override;
+
+  void HandleGetBraveRewardsEnabled(const base::Value::List& args);
+
+  void HandleClearBraveAdsData(const base::Value::List& args);
+
+  void UpdateBraveRewardsEnabled();
```
Comment (tmancey) [inline browser/ui/webui/settings/brave_clear_browsing_data_handler.h:33]: Maybe choose a different name, because the function does not update anything.

Comment (tmancey) [reply browser/ui/webui/settings/brave_clear_browsing_data_handler.h:33]: Maybe... `OnRewardsEnabledPreferenceChanged`

**Review by tmancey — APPROVED** [review]
LGTM++ (Please see comments)

File: ios/brave-ios/Sources/BraveStrings/BraveStrings.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (aseren) [reply ios/brave-ios/Sources/BraveStrings/BraveStrings.swift]: Changed to toggle text to `Brave Ads Data`

File: components/brave_rewards/browser/rewards_service_impl.cc (hunk 6 above)
Comment (aseren) [reply components/brave_rewards/browser/rewards_service_impl.cc]: Agree that this is an edge case. Removed this setting pref change above:
`profile_->GetPrefs()->SetBoolean(brave_rewards::prefs::kEnabled, false);`

File: ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift (hunk 1 above)
Comment (aseren) [reply ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift]: added @MainActor

File: ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController.swift (hunk 9 above)
Comment (aseren) [reply ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController.swift]: Fixed

File: components/brave_ads/browser/ads_service_impl.cc (hunk 4 above)
Comment (aseren) [reply components/brave_ads/browser/ads_service_impl.cc]: The concern is fixed by adding `Clear Brave Ads data...` link to the settings for non-Rewards users. 
When clicking this link all Brave Ads files and preferences are deleted.

If user joined Brave Rewards then `Clear Brave Ads data...` link is replaced by `Reset Brave Rewards data...`, which reset rewards and clears all Brave Ads files and preferences as well.

File: components/brave_ads/core/internal/history/database_history_util_internal.cc (hunk 5 above)
Comment (aseren) [reply components/brave_ads/core/internal/history/database_history_util_internal.cc]: The concern is fixed by adding `Clear Brave Ads data...` link to the settings for non-Rewards users. 
When clicking this link all Brave Ads files and preferences are deleted.

If user joined Brave Rewards then `Clear Brave Ads data...` link is replaced by `Reset Brave Rewards data...`, which allows to reset rewards and clear all Brave Ads files and preferences as well.

File: ios/browser/api/ads/brave_ads.mm (hunk 10 above)
Comment (aseren) [reply ios/browser/api/ads/brave_ads.mm:1759]: Fixed

File: chromium_src/chrome/browser/ui/webui/settings/settings_ui.cc [hunk 12]
```diff
@@ -20,7 +21,9 @@
 #define SiteSettingsHandler BraveSiteSettingsHandler
 #define ImportDataHandler BraveImportDataHandler
 #define SearchEnginesHandler BraveSearchEnginesHandler
+#define ClearBrowsingDataHandler BraveClearBrowsingDataHandler
```
Comment (mkarolin) [inline chromium_src/chrome/browser/ui/webui/settings/settings_ui.cc:24]: Should probably also #include "chrome/browser/ui/webui/settings/settings_clear_browsing_data_handler.h" so that `ClearBrowsingDataHandler` isn't redefined in there.

Comment (mkarolin) [reply chromium_src/chrome/browser/ui/webui/settings/settings_ui.cc:24]: Although, I guess browser/ui/webui/settings/brave_clear_browsing_data_handler.h already include it.

**Review by mkarolin — APPROVED** [review]
`chromium_src` and `strings` ++

**Review by zenparsing — APPROVED** [review]

File: ios/browser/api/ads/brave_ads.mm [hunk 13]
```diff
@@ -1739,6 +1740,31 @@ - (void)toggleDislikeAd:(NSString*)creativeInstanceId
                        /*intentional*/ base::DoNothing());
 }
 
+- (void)clearData:(void (^)())completion {
+  // Ensure the Brave Ads service is stopped before clearing data.
+  if ([self isServiceRunning]) {
+    return completion();
```
Comment (kylehickinson) [inline ios/browser/api/ads/brave_ads.mm:1746]: Is it valid to call this while its running? If not perhaps we should add a `DCHECK` here

File: ios/brave-ios/Sources/BraveStrings/BraveStrings.swift [hunk 14]
```diff
@@ -9183,6 +9183,16 @@ extension Strings {
...
     )
+
+    public static let braveAdsDataToggleOption =
+      NSLocalizedString(
+        "ads.braveAdsDataToggleOption",
+        tableName: "BraveShared",
+        bundle: .module,
+        value: "Brave Ads Data",
+        comment:
+          "Text for Brave Ads Data Toggle while clearing the data in settings"
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/BraveStrings/BraveStrings.swift:9194]: Is "Ads" here supposed to be translated? Can you be specific in the comment for the translator to ensure that at least "Brave" or "Brave Ads" is the product name and shouldn't be translated and that data refers to information stored on device

File: ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift [hunk 15]
```diff
@@ -192,6 +200,29 @@ public class BraveRewards: PreferencesObserver {
...
+    await withCheckedContinuation { continuation in
+      ads.shutdownService {
+        continuation.resume()
+      }
+    }
+    await withCheckedContinuation { continuation in
+      ads.clearData {
+        continuation.resume()
+      }
+    }
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift:216]: You can simplify this to 
```swift
await withCheckedContinuation { continuation in
  ads.shutdownService {
    ads.clearData {
      continuation.resume()
    }
  }
}
```

File: ios/brave-ios/Sources/BraveStrings/BraveStrings.swift (hunk 14 above)
Comment (aseren) [reply ios/brave-ios/Sources/BraveStrings/BraveStrings.swift:9194]: `Brave Ads` shouldn't be translated. I've changed comment to `Label for the toggle switch to clear 'Brave Ads' data in settings`

File: ios/browser/api/ads/brave_ads.mm (hunk 13 above)
Comment (aseren) [reply ios/browser/api/ads/brave_ads.mm:1746]: Changed if statement to `CHECK(![self isServiceRunning]);`

**Review by kylehickinson — APPROVED** [review]
